client/blocks/login/social.jsx

```diff
@@ -55,6 +61,8 @@ class SocialLoginForm extends Component {
 									error_message: createAccountError.message
 								} )
 							);
+					} else if ( error.code === 'existing_wpcom_user' ) {
+						this.props.linkSocialUser( 'google', error.email );
 					}
 
 					this.recordEvent( 'calypso_login_social_login_failure', {
```

Comment:
Not related to this pull request, but we should log the `social_account_type` property in all the events triggered by `handleGoogleResponse()`.